### PR TITLE
Finish pr 178

### PR DESCRIPTION
**Pull request by rcqz**
This PR resolves the conversations from PR #178 and refactors the code. Some additional tests are also added.

## Summary by CodeRabbit

- **New Features**
  - Introduced four new models to estimate work capacity under heat stress: Dunne et al., Hothaps, ISO, and NIOSH.
  - Added support for specifying work intensity levels and metabolic rate in work capacity calculations.
  - Added a new result class to represent work capacity as a percentage.

- **Documentation**
  - Extended documentation with detailed sections for each new work capacity model.
  - Added relevant scientific references related to heat stress and labor capacity.

- **Tests**
  - Added comprehensive tests for all new work capacity models, covering typical scenarios, boundary conditions, and invalid inputs.

### CODE REVIEW

**Comment by FedericoTartarini** [comment]
@rcqz please send me an email once the pull request has been completed, all the conflicts are merged and you have addressed and reviewed the code rabbit comments.

**Comment by rcqz** [comment]
@coderabbitai review

File: pythermalcomfort/classes_input.py [hunk 1]
```diff
@@ -53,6 +61,8 @@ class BaseInputs:
     max_skin_blood_flow: Union[float, int, np.ndarray, list] = field(default=80)
     max_sweating: Union[float, int, np.ndarray, list] = field(default=500)
     w_max: Union[float, int, np.ndarray, list] = field(default=None)
+    wbgt: Union[float, int, np.ndarray, list] = field(default=None)
+    intensity: Union[str, WorkIntensity] = field(default=None)
```
Comment (FedericoTartarini) [inline pythermalcomfort/classes_input.py]: @rcqz is this work intensity? If so we should call it work_intensity.

File: pythermalcomfort/classes_input.py [hunk 2]
```diff
@@ -7,6 +7,14 @@
 from pythermalcomfort.utilities import Postures, Sex, Units, validate_type
 
 
+class WorkIntensity(str, Enum):
```
Comment (FedericoTartarini) [inline pythermalcomfort/classes_input.py:10]: @rcqz is this correct or should it be `class WorkIntensity(Enum):`?

Comment (rcqz) [reply pythermalcomfort/classes_input.py:10]: I think including str is fine because this allows the members to behave like strings. Otherwise we will have to do the value attribute every time

**Review by FedericoTartarini — APPROVED** [review]
